user_interface/EbikeUIController/can.cpp

```diff
@@ -0,0 +1,121 @@
+}
+
+//function used as callback when bike status message enters mailbox
+void updateCAN_BikeStatus_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables  
+    CAN_BikeStatus_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_BikeStatus_Struct.speed = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_BikeStatus_Struct.temp = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_BikeStatus_Struct.throttle = msg.buf[6];
+    CAN_BikeStatus_Struct.error = msg.buf[7];
+    CAN_MB0_BikeStatus_Flag = true; // updated values have been read
+}
+
+//function used as callback when motor voltage message enters mailbox
+void updateCAN_MotorVoltages_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_MotorVoltages_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_MotorVoltages_Struct.vDC = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_MotorVoltages_Struct.vA = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_MotorVoltages_Struct.vB = (msg.buf[6] << 8) | msg.buf[7];
+    CAN_MB1_MotorVoltages_Flag = true; // updated values have been read
+}
+
+//function used as callback when real current message enters mailbox
+void updateCAN_RealCurrents_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_RealCurrents_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_RealCurrents_Struct.iDC = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_RealCurrents_Struct.iA = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_RealCurrents_Struct.iB = (msg.buf[6] << 8) | msg.buf[7];
+    CAN_MB2_RealCurrents_Flag = true; // updated values have been read
+}
+
+//function used as callback when calculated values message enters mailbox
+void updateCAN_CalcValues_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_CalcValues_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_CalcValues_Struct.iqRef = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_CalcValues_Struct.iqFdb = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_CalcValues_Struct.power = (msg.buf[6] << 8) | msg.buf[7];
+    CAN_MB3_CalcValues_Flag = true; // updated values have been read
+}
+
+//function used as callback when BMS message enters mailbox
+void updateCAN_BmsSoc_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_BmsSoc_Struct.soc = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_BmsSoc_Struct.soh = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_MB4_BmsSoc_Flag = true; // updated values have been read
+}
+
+// gets the received values from the CAN message and stores values in main
+void getCAN_BikeStatus_Values(uint16_t &timestamp, uint16_t &Speed, uint16_t &temp, uint8_t &throttleInput, uint8_t &ErrorWarning) {
+    timestamp = CAN_BikeStatus_Struct.timestamp; 
+    Speed = CAN_BikeStatus_Struct.speed;
+    temp = CAN_BikeStatus_Struct.temp_fet;
+    throttleInput = CAN_BikeStatus_Struct.throttleInput;
+    ErrorWarning = CAN_BikeStatus_Struct.errorWarning; 
```

Comment:
To make the interface simpler, I think it would be good to reset the message flags at the end of these functions. That way, after reading a message the user doesn't have to manually reset the flag. In the main file, the final interface would simplify to something like:

`if (canPendingBikeStatusMsg()){
    getCAN_BikeStatus_Values(...);
}`